physics/pace/physics/stencils/physics.py

```diff
@@ -249,8 +248,8 @@ def make_quantity():
                 "pktop": self._pktop,
             },
         )
-        if "microphysics" in active_packages:
-            self._do_microphysics = True
+        if "GFS_microphysics" in schemes:
+            self._gfs_microphysics = True
```

Comment:
Shouldn't there be a more "Pythonic" way of doing this, like with dictionaries or something, instead of a long tree of `if` statements?

Reply:
IMO, this isn't as much as Pythonic need rather than a full OO structure with ABC class at the top and overrides going down. With a registrar that registers Physics packages with the same public API

Reply:
This is a very good point, and we definitely want to use a more OO design for this, but I'm hesitant to implement it fully until I have a fuller picture of what that will require to cover all of the physics schemes.

ALTHOUGH, if we're splitting things up as part of the model/framework refactor, do we actually want Pace to support different physics configs like this? This PR is mostly to smooth out issues that will crop up once we have the SHiELD microphysics as an option in addition to the GFS microphysics, but does multi-scheme support even make sense for a single model like Pace?

Reply:
Both SHiELD and AM4/AM5 do support a selected number of alternative parameterizations; SHiELD carries three PBL schemes, for instance, and AM5 gives a choice of several microphysics and convection schemes. Pace doesn't need to support every single Legacy scheme, and we certainly don't want to bring in dozens of schemes or suites like WRF or CCPP, but at least for development purposes we should allow at least some runtime options for physical parameterizations.

Reply:
From a software architecture standpoint this sounds like plug & play API + sets of reasonable defaults.

Reply:
Yeah that's really helpful to know, thanks